pkg/app/piped/executor/kubernetes/applier_group.go

```diff
@@ -46,48 +49,99 @@ func newApplierGroup(defaultProvider string, appCfg config.KubernetesApplication
 		logger,
 	)
 	d := &applierGroup{
-		resourceRoutes: appCfg.ResourceRoutes,
-		appliers:       map[string]provider.Applier{defaultProvider: defaultApplier},
-		defaultApplier: defaultApplier,
+		resourceRoutes:   appCfg.ResourceRoutes,
+		appliers:         map[string]provider.Applier{defaultProvider: defaultApplier},
+		labeledProviders: make(map[string][]string, 0),
+		defaultApplier:   defaultApplier,
 	}
 
 	for _, r := range appCfg.ResourceRoutes {
-		providerName := r.Provider.Name
-		if _, ok := d.appliers[providerName]; ok {
+		if name := r.Provider.Name; name != "" {
+			if _, ok := d.appliers[name]; ok {
+				continue
+			}
+			cp, ok := pipedCfg.FindPlatformProvider(name, model.ApplicationKind_KUBERNETES)
+			if !ok {
+				return nil, fmt.Errorf("provider %s specified in resourceRoutes was not found", name)
+			}
+			d.appliers[name] = provider.NewApplier(appCfg.Input, *cp.KubernetesConfig, logger)
 			continue
 		}
-
-		cp, ok := pipedCfg.FindPlatformProvider(providerName, model.ApplicationKind_KUBERNETES)
-		if !ok {
-			return nil, fmt.Errorf("provider %s specified in resourceRoutes was not found", providerName)
+		if labels := r.Provider.Labels; len(labels) > 0 {
```

Comment:
Sorry, Let me check one point.
Is it possible for a user to identify a specific platform provider without that name?

Reply:
Yes, I assume some use cases as below:
- using only name to specify exactly one provider
- using labels without name to specify multiple providers

So it is possible to give only the labels without name.
Wdyt?

Reply:
That sounds good! Thank you. 🙌